spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/integration/IntegrationProperties.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2012-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.integration;
+
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.integration.http.support.HttpContextUtils;
+
+/**
+ * Configuration properties for Spring Integration.
+ *
+ * @author Artem Bilan
+ * @since 1.4
+ * @see org.springframework.integration.context.IntegrationProperties
+ */
+@ConfigurationProperties("spring.integration")
+public class IntegrationProperties {
+
+	private final Channels channels = new Channels();
+
+	/**
+	 * The number of threads available in the default taskScheduler.
```

Comment:
Thank you all for all the feedback!
I'll try to follow with your requirements and will go ahead with short steps.

Please, join me to that discussion about default `TaskScheduler`. We definitely should consider to reuse that auto-configured for Spring Integration as well. Although that might be just like copy/paste options because our default is like : 
```
BeanDefinition scheduler = BeanDefinitionBuilder.genericBeanDefinition(ThreadPoolTaskScheduler.class)
		.addPropertyValue("poolSize", IntegrationProperties.getExpressionFor(IntegrationProperties.TASK_SCHEDULER_POOL_SIZE))
		.addPropertyValue("threadNamePrefix", "task-scheduler-")
		.addPropertyValue("rejectedExecutionHandler", new CallerRunsPolicy())
		.addPropertyValue("errorHandler", new RootBeanDefinition(MessagePublishingErrorHandler.class))
		.getBeanDefinition();
```

The most important option is `MessagePublishingErrorHandler`.

OK. Never mind. We can discuss that on its own turn.